ffmpeg/lpms_ffmpeg.c

```diff
@@ -743,6 +743,16 @@ static int open_video_decoder(input_params *params, struct input_ctx *ctx)
   AVCodec *codec = NULL;
   AVFormatContext *ic = ctx->ic;
 
+  if (AV_HWDEVICE_TYPE_CUDA == params->hw_type) {
+    for( int i = 0; i < ic->nb_streams; i++ ) {
+      if( ic->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO &&
```

Comment:
We already do this type of check a few lines down. `ctx->vi` is the index of the video stream we'll be working with.

See this block:

https://github.com/livepeer/lpms/blob/29167f81e36cd21d20d53619332f8a4bf1a50695/ffmpeg/lpms_ffmpeg.c#L762-L767

Also, `AVStream.codec` is deprecated; did the compiler not warn about it? See https://github.com/FFmpeg/FFmpeg/blob/343ccfcc4deb2257e1525f05284e6343816a91cc/libavformat/avformat.h#L879-L885

That being said, this should hopefully do the trick:

```diff
diff --git a/ffmpeg/lpms_ffmpeg.c b/ffmpeg/lpms_ffmpeg.c
index 76e30db..a57cff1 100644
--- a/ffmpeg/lpms_ffmpeg.c
+++ b/ffmpeg/lpms_ffmpeg.c
@@ -754,6 +754,10 @@ static int open_video_decoder(input_params *params, struct input_ctx *ctx)
       AVCodec *c = avcodec_find_decoder_by_name("h264_cuvid");
       if (c) codec = c;
       else fprintf(stderr, "Cuvid decoder not found; defaulting to software\n");
+      if (AV_PIX_FMT_YUV420P != ic->streams[ctx->vi]->codecpar->format) {
+        ret = lpms_ERR_INPUT_PIXFMT;
+        dd_err("Non 4:2:0 pixel format detected in input");
+      }
     }
     AVCodecContext *vc = avcodec_alloc_context3(codec);
     if (!vc) dd_err("Unable to alloc video codec\n");
```